metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java

```diff
@@ -126,6 +87,27 @@ public HttpRecordIterator harvestRecords(String archiveUrl, String downloadDirec
     return harvestRecords(downloadedFile);
   }
 
+  @Override
+  public HttpRecordIterator createTemporaryHttpHarvestIterator(InputStream input, CompressedFileExtension compressedFileType) throws HarvesterException {
+    // We chose where to store the temporary file.
+    @SuppressWarnings("findsecbugs:PATH_TRAVERSAL_IN")
+    Path tempDir;
+    final Path tempFile;
+    try {
+      // Save the zip file in a temporary directory (and close the input stream).
+      final String prefix = UUID.randomUUID().toString();
+
+      tempDir = Files.createTempDirectory(prefix);
+      tempFile = Files.createTempFile(tempDir, prefix, compressedFileType.getExtension());
+      FileUtils.copyInputStreamToFile(input, tempFile.toFile());
+
+      return harvestRecords(tempFile);
+      } catch (IOException e) {
+        throw new HarvesterException("Problem saving archive.", e);
+      }
+
```

Comment:
close ```input``` stream to prevent memory leakage. if applies use try-with-resources to close stream or do it manually.

Reply:
The input stream is closed insdie the method copyInputStreamToFile()